plugin/kubernetes/metrics.go

```diff
@@ -27,7 +23,7 @@ var (
 	//   * headless_without_selector
 	DnsProgrammingLatency = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: plugin.Namespace,
-		Subsystem: subsystem,
+		Subsystem: "kubernetes",
```

Comment:
It's typically preferred to make these a const/var to avoid copy-paste mistakes when there are multiple metrics.

I might suggest we have a standard name like `pluginName` (doesn't need to be exported from the plugin package. This would make it easier and more clear than just having it documented as a must.